p2p/chain.py

```diff
@@ -193,28 +174,25 @@ def register_peer(self, peer: BasePeer) -> None:
         # find the common ancestor between our chain and the peer's.
         start_at = max(0, head.block_number - eth.MAX_HEADERS_FETCH)
         while not self._sync_complete.is_set():
-            if peer.is_finished:
+            if not peer.is_running:
```

Comment:
Why the inversion?  The `not` seems to make this slightly harder to read.

Reply:
Because I got rid of the `self.finished` event and I don't think it makes sense to have both `is_running` and `is_finished` properties.